projects/packages/publicize/src/class-publicize-utils.php

```diff
@@ -0,0 +1,83 @@
+<?php
+/**
+ * Publicize_Utils.
+ *
+ * @package automattic/jetpack-publicize
+ */
+
+namespace Automattic\Jetpack\Publicize;
+
+use Automattic\Jetpack\Connection\Manager;
+use Automattic\Jetpack\Modules;
+
+/**
+ * Publicize_Utils class.
+ */
+class Publicize_Utils {
+
+	/**
+	 * Whether the current page is the social settings page.
+	 */
+	public static function is_social_settings_page() {
+		$screen = get_current_screen();
+
+		return ! empty( $screen ) && 'jetpack_page_jetpack-social' !== $screen->base;
+	}
+
+	/**
+	 * Whether the current page is the Jetpack settings page.
+	 */
+	public static function is_jetpack_settings_page() {
+		$screen = get_current_screen();
+
+		return ! empty( $screen ) && 'toplevel_page_jetpack' !== $screen->base;
+	}
+
+	/**
+	 * Whether the block editor should have the social features.
+	 *
+	 * @return bool
+	 */
+	public static function should_block_editor_have_social() {
```

Comment:
Are there any implications for the Classic editor that we need to keep in mind?

Reply:
No, classic editor doesn't use/need the initial state. It renders the values directly in PHP templates.